base/div.jl

```diff
@@ -10,7 +10,11 @@ an integer according to the rounding mode `r`. In other words, the quantity
 
     round(x / y, r)
 
-without any intermediate rounding.
+without any intermediate rounding. `div` is faster when `x` and `y` are 32-bit integers,
+than for 64-bit, but note even smaller operands, like 16-bit, are not faster for the
+operation (speed can though increase indirectly with smaller sizes, since memory traffic
```

Comment:
Isn't that highly dependent on the target architecture? Did you verify this on 32bit and 64 bit, x86 and ARM?

Reply:
I do see:
https://developer.arm.com/documentation/den0024/a/The-A64-instruction-set/Data-processing-instructions/Multiply-and-divide-instructions
>The ARMv8-A architecture has support for signed and unsigned division of 32-bit and 64-bit sized values.

Maybe someone can time for me.

I only timed on x86-64. It's hypothetical if your 64-bit values fit into a 32-bit register, then as fast for integers, since then no less complex, if the uarch would do that, but I didn't confirm same speed on my CPU. For 32-bit float division it's however inherently more complex.